lib/fusion.flow

```diff
@@ -1530,44 +1530,26 @@ fmin3(b1 : Transform<?>, b2 : Transform<?>, b3 : Transform<?>) -> Transform<?> {
 
 fmaxA(v : [Transform<?>], def : ?) -> Transform<?> {
 	if (length(v) > 0)
-		foldTreeBinary(
-			pairs2tree(mapi(v, \k : int, vv : Transform<?> -> Pair(k, Some(vv)))),
-			\__ : int, vv : Maybe<Transform<?>>, ll : Maybe<Transform<?>>, rr : Maybe<Transform<?>> -> {
-				maybeMap(vv, \vvv -> eitherFn(ll,
-					\lll -> {
-						m = fmax(vvv, lll);
-						eitherMap(rr, \rrr -> fmax(m, rrr), m)
-					},
-					\ -> eitherMap(rr, \rrr -> fmax(vvv, rrr), vvv)
-				))
-			},
-			None()
-		)
-		|> (\f -> either(f, const(def)))
+		fMaxMin(v, fmax)
 	else
 		ConstBehaviour(def)
 }
 
 fminA(v : [Transform<?>], def : ?) -> Transform<?> {
 	if (length(v) > 0)
-		foldTreeBinary(
-			pairs2tree(mapi(v, \k : int, vv : Transform<?> -> Pair(k, Some(vv)))),
-			\__ : int, vv : Maybe<Transform<?>>, ll : Maybe<Transform<?>>, rr : Maybe<Transform<?>> -> {
-				maybeMap(vv, \vvv -> eitherFn(ll,
-					\lll -> {
-						m = fmin(vvv, lll);
-						eitherMap(rr, \rrr -> fmin(m, rrr), m)
-					},
-					\ -> eitherMap(rr, \rrr -> fmin(vvv, rrr), vvv)
-				))
-			},
-			None()
-		)
-		|> (\f -> either(f, const(def)))
+		fMaxMin(v, fmin)
 	else
 		ConstBehaviour(def)
 }
 
+fMaxMin(v : [Transform<?>], fn : (Transform<?>, Transform<?>) -> Transform<?>) -> Transform<?> {
```

Comment:
I think it can be reworked without `subrange`, just pass left and right indexes to the function.

Reply:
done